### ISSUE TITLE
App crashed when i opened my shopping list

### ISSUE THREAD

**Issue by gergelypolonkai** [stefan-niedermann/nextcloud-notes#994]
Please use **[GitHub reactions](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** 👍 to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!

**Describe the bug**

When i opened my shopping list note (a document with some headings, checkboxed lists, and a bunch of emojis), the app crashed.

**To Reproduce**

I cannot willfully reproduce the issue. It happens *sometimes*.

**Expected behavior**

The app should have opened my note.

**Screenshots**

N/A

**Smartphone (please complete the following information):**
 - Nextcloud Notes-Version (android app): 2.17.1
 - F-Droid or Play Store: F-Hroid
 - Android-Version: 10
 - Device: Huawei P20 Pro

**Server**
 - Nextcloud version: 19. 0.5
 - Nextcloud Notes version (server app): 3.6.4

**Stacktrace**

```
App Version: 2.17.1
App Version Code: 2017001
App Flavor: fdroid

Files App Version Code: 30130190

---

OS Version: 4.14.116(10.0.0.171C432)
OS API Level: 29
Device: HWCLT
Manufacturer: HUAWEI
Model (and Product): CLT-L29 (CLT-L29)

---

java.lang.ArrayIndexOutOfBoundsException: length=244; index=-1
	at java.util.ArrayList.get(ArrayList.java:439)
	at it.niedermann.owncloud.notes.main.items.ItemAdapter.getItem(ItemAdapter.java:202)
	at it.niedermann.owncloud.notes.main.MainActivity.onNoteClick(MainActivity.java:867)
	at it.niedermann.owncloud.notes.main.items.NoteViewHolder.lambda$bind$0$NoteViewHolder(NoteViewHolder.java:53)
	at it.niedermann.owncloud.notes.main.items.-$$Lambda$NoteViewHolder$USpBz4_HPjUNfVR-HTVpNrLSa0I.onClick(Unknown Source:2)
	at android.view.View.performClick(View.java:7192)
	at android.view.View.performClickInternal(View.java:7166)
	at android.view.View.access$3500(View.java:824)
	at android.view.View$PerformClick.run(View.java:27592)
	at android.os.Handler.handleCallback(Handler.java:888)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:213)
	at android.app.ActivityThread.main(ActivityThread.java:8178)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1101)
```

**Comment by markushub** [comment]
```
App Version: 2.17.1
App Version Code: 2017001
App Flavor: fdroid

Files App Version Code: 30140090

---

OS Version: 3.18.120-perf-Arrow(2048)
OS API Level: 29
Device: potter
Manufacturer: motorola
Model (and Product): Moto G5 Plus (XT1685) (Moto G5 Plus)

---

java.lang.ArrayIndexOutOfBoundsException: length=33; index=-1
	at java.util.ArrayList.get(ArrayList.java:439)
	at it.niedermann.owncloud.notes.main.items.ItemAdapter.getItem(ItemAdapter.java:202)
	at it.niedermann.owncloud.notes.main.MainActivity.onNoteClick(MainActivity.java:867)
	at it.niedermann.owncloud.notes.main.items.NoteViewHolder.lambda$bind$0$NoteViewHolder(NoteViewHolder.java:53)
	at it.niedermann.owncloud.notes.main.items.-$$Lambda$NoteViewHolder$USpBz4_HPjUNfVR-HTVpNrLSa0I.onClick(Unknown Source:2)
	at android.view.View.performClick(View.java:7259)
	at android.view.View.performClickInternal(View.java:7236)
	at android.view.View.access$3600(View.java:801)
	at android.view.View$PerformClick.run(View.java:27892)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7356)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:940)

```

**Comment by stefan-niedermann** [comment]
Fix will be released with `v2.17.2` - thank you very much @Isaac-Graham !
